addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -490,6 +500,22 @@ options.registry.gallery = options.Class.extend({
     _getColumns: function () {
         return parseInt(this.$target.attr('data-columns')) || 3;
     },
+    /**
+     * Returns the images (or the images wrapped in their anchor if they were
+     * linked) sorted by index.
+     *
+     * @private
+     * @returns {DOMElement[]}
+     */
+    _getLinkedImages: function () {
+        const imgEls = this._getImages();
+        return imgEls.map((imgEl) => {
+            if (imgEl.parentElement.tagName === "A") {
+                return imgEl.parentElement;
+            }
+            return imgEl;
+        });
```

Comment:
```suggestion
        return imgEls.map(imgEl => imgEl.closest('a') || imgEl);
```
One line and you actually handle the case where the images are not the direct children of the links.